.github/workflows/ci.yml

```diff
@@ -109,10 +109,108 @@ jobs:
           COMMIT_MESSAGE: ${{ github.event.head_commit.message }}
           COMMIT_AUTHOR: ${{ github.event.head_commit.author.username }}
 
+  build-base:
+    name: base (${{ matrix.image }})
+    runs-on: ubuntu-latest
+    needs: [shellcheck, test, check]
+    if: github.event_name == 'push'
+    strategy:
+      fail-fast: false
+      matrix:
+        image:
+          - base
+        sub:
+          - ubuntu
+          - centos
+          - emscripten
+    outputs:
+      coverage-artifact: ${{ steps.cov.outputs.artifact-name }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: ./.github/actions/setup-rust
+
+      - name: Set up Docker Buildx
+        if: runner.os == 'Linux'
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build xtask
+        run: cargo build -p xtask
+
+      - name: Prepare Meta
+        id: prepare-meta
+        timeout-minutes: 60
+        run: cargo xtask ci-job prepare-meta "${IMAGE}.${SUB}"
+        env:
+          IMAGE: ${{ matrix.image }}
+          SUB: ${{ matrix.sub }}
+        shell: bash
+
+      - name: LLVM instrument coverage
+        uses: ./.github/actions/cargo-llvm-cov
+        with:
+          name: cross-${{matrix.image}}-${{matrix.sub}}
+
+      - name: Install cross
+        if: matrix.deploy
+        run: cargo install --path . --force --debug
+
+      - name: Docker Meta
+        id: docker-meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            name=${{ steps.prepare-meta.outputs.image }}
+          labels: |
+            ${{ fromJSON(steps.prepare-meta.outputs.labels) }}
+      - name: Build Docker image
+        id: build-docker-image
+        timeout-minutes: 60
+        run: cargo xtask build-docker-image -v "${IMAGE}.${SUB}"
+        env:
+          IMAGE: ${{ matrix.image }}
+          SUB: ${{ matrix.sub }}
+          LABELS: ${{ steps.docker-meta.outputs.labels }}
+          LATEST: ${{ needs.check.outputs.is-latest || 'false' }}
+        shell: bash
+
+      - name: Save Docker Image
+        id: save-docker-image
+        run: docker save "${IMAGE}:${TAG}" -o "${IMAGE}-${TAG}-${SUB}.tar"
```

Comment:
Our `docker load` step always needs to load these as `main`, since that's what we use for our base images:

```Dockerfile
FROM ghcr.io/cross-rs/base:emscripten-main
```

We technically don't have to deploy these images, but it might be nice for people trying to build new images based off this logic locally, especially cross maintainers.